doc/api/packages.md

```diff
@@ -819,10 +737,219 @@ conditional exports for consumers could be to add an export, e.g.
 }
 ```
 
+## `package.json` supported fields
+
+This document aims to describe the fields used by the Node.js
+runtime. Other tools (such as
+[npm](https://docs.npmjs.com/creating-a-package-json-file)) may use additional
+fields which are ignored by Node.js and not documented here.
+
+### `"name"`
+<!-- YAML
+added:
+  - v12.16.0
+  - v13.1.0
+changes:
+  - version:
+    - v12.16.0
+    - v13.6.0
+    pr-url: https://github.com/nodejs/node/pull/31002
+    description: Remove the `--experimental-resolve-self` option.
+-->
+
+```json
+{
+  "name": "package-name"
+}
+```
+
+* Type: {string}
+
+The `"name"` field defines your package’s name. Node.js doesn't apply any
+restriction on the name field, although the field is ignored if it is not a
+string or an empty string.
```

Comment:
I've added the link to npm docs, PTAL

Reply:
I'd still prefer to remove this sentence -  "Node.js doesn't apply any
restriction on the name field, although the field is ignored if it is not a
string or an empty string."

Reply:
OK, I personally feel we should have documented somewhere what are the limits Node.js enforces. Are you saying I should rephrase it or I should drop it because it's not relevant?

Reply:
Right, the thing is it is not quite this simple - because when loading self-resolution it must match the package name pattern of `/^(@[^@\/]+]\/)?[^@\/]+$/` and I'm not sure we should get into that here.

Reply:
Interesting, I didn't know about the pattern! Goeffrey also agrees the sentence is irrelevant (https://github.com/nodejs/node/pull/34970#discussion_r483296692), I'm removing it.